modules/objdetect/src/qrcode.cpp

```diff
@@ -1001,217 +1070,1329 @@ void QRDecode::init(const Mat &src, const vector<Point2f> &points)
+                }
+            }
+            else if (index_p1_in_contour < index_p2_in_contour)
+            {
+                for (size_t k = index_p1_in_contour; k <= index_p2_in_contour; k++)
+                {
+                    points.push_back(add_points[k]);
+                }
+            }
+            else { return false; }
+            if (abs(p1.x - p2.x) > abs(p1.y - p2.y))
+            {
+                sort(points.begin(), points.end(), sortPointsByX());
+            }
+            else
+            {
+                sort(points.begin(), points.end(), sortPointsByY());
+            }
+
+            temp_patterns_add_points.push_back(std::pair<int, vector<Point> >(idx_curved_side,points));
+        }
+    }
+
+    return true;
+}
+
+bool QRDecode::computePatternsAddingPoints(std::map<int, vector<Point> > &patterns_add_points)
+{
+    vector<std::pair<int, vector<Point> > > temp_patterns_add_points;
+    if(!findTempPatternsAddingPoints(temp_patterns_add_points)) { return false; }
+
+    const int num_points_in_pattern = 3;
+    for(size_t i = 0; i < temp_patterns_add_points.size(); i++)
+    {
+        int idx_side = temp_patterns_add_points[i].first;
+        int size = (int)temp_patterns_add_points[i].second.size();
+
+        float step = static_cast<float>(size) / num_points_in_pattern;
+        vector<Point> temp_points;
+        for (int j = 0; j < num_points_in_pattern; j++)
+        {
+            float val = j * step;
+            int idx = cvRound(val) >= size ? size - 1 : cvRound(val);
+            temp_points.push_back(temp_patterns_add_points[i].second[idx]);
+        }
+        temp_points.push_back(temp_patterns_add_points[i].second.back());
+        if(patterns_add_points.count(idx_side) == 1)
+        {
+            patterns_add_points[idx_side].insert(patterns_add_points[idx_side].end(),
+                                                temp_points.begin(), temp_points.end());
+        }
+        patterns_add_points.insert(std::pair<int, vector<Point> >(idx_side, temp_points));
+
+    }
+    if (patterns_add_points.size() == 0) { return false;}
+
+    return true;
+}
+
+bool QRDecode::addPointsToSides()
+{
+    if(!computePatternsAddingPoints(complete_curved_sides)) { return false; }
+
+    std::map<int, vector<Point> >::iterator it;
+    double mean_step = 0.0;
+    size_t num_points_at_side = 0;
+    for (it = complete_curved_sides.begin(); it != complete_curved_sides.end(); ++it)
+    {
+        int count = -1;
+        const size_t num_points_at_pattern = it->second.size();
+        for(size_t j = 0; j < num_points_at_pattern - 1; j++, count++)
+        {
+            if (count == 3) continue;
+            double temp_norm = norm(it->second[j] -
+                                    it->second[j + 1]);
+            mean_step += temp_norm;
+        }
+        num_points_at_side += num_points_at_pattern;
+    }
+    if (num_points_at_side == 0) { return false; }
```

Comment:
Please use separate line for `if` body, even for simple cases.
It is important to see the real code coverage.